strawberryfields/tdm/tdmprogram.py

```diff
@@ -554,9 +551,16 @@ def parameters(self):
     def roll(self):
         """Represent the program in a compressed way without rolling the for loops"""
         self.circuit = self.rolled_circuit
+        if self._is_space_unrolled:
+            if self.added_subsystems > 0:
+                self._delete_subsystems(self.register[-self.added_subsystems :])
+                self.init_num_subsystems -= self.added_subsystems
+                self.added_subsystems = 0
+
+            self._is_space_unrolled = False
         return self
 
-    def unroll(self, shots):
+    def unroll(self, shots=1):
```

Comment:
How come `shots=1` is the default now?

Reply:
It seems to be the most common case (at least when looking at the usage so far), and I believe it makes sense having this as default. There where a lot of `prog.unroll(1)` showing up where the `1` seemed to confused some people, which was partly my fault for writing it like that at some point. 🙂